pkgs/tools/admin/ansible/default.nix

```
@@ -5,7 +5,7 @@ rec {
 
   ansible_2_9 = python3Packages.toPythonApplication python3Packages.ansible;
 
-  ansible_2_8 = with python3Packages; toPythonApplication (python3Packages.ansible.overrideAttrs(old: rec {
+  ansible_2_8 = with python3Packages; toPythonApplication (python3Packages.ansible.overridePythonAttrs (old: rec {
```

Why not this:
```suggestion
  ansible_2_8 = with python3Packages; toPythonApplication (ansible.overridePythonAttrs (old: rec {
```

---

I guess could be this:
```suggestion
  ansible_2_8 = python3Packages.toPythonApplication (python3Packages.ansible.overridePythonAttrs (old: rec {
```
because
```
nix-repl> with { x = 5; }; with { x = 6; }; x
6

nix-repl> let x = 5; in with { x = 6; }; x    
5

nix-repl> (rec { x = 5; y = with { x = 6; }; x; }).y
5
```
But because that asks to incorrectly refactor it, we could remove `ansible` attribute from the `rec {}` and define it in the all-packages.nix instead. Then `with python3Packages;` will work fine.

---

You’ve discovered why your first suggestion doesn’t work. I’ve removed the `with` as in your second suggestion.